app/src/main/java/ch/epfl/sdp/drone3d/service/module/LocationModule.kt

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (C) 2021  Drone3D-Team
+ * The license can be found in LICENSE at root of the repository
+ */
+
+package ch.epfl.sdp.drone3d.service.module
+
+import android.content.Context
+import android.location.Criteria
+import android.location.LocationManager
+import ch.epfl.sdp.drone3d.service.api.location.LocationService
+import ch.epfl.sdp.drone3d.service.impl.location.AndroidLocationService
+import dagger.Binds
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.android.qualifiers.ApplicationContext
+import dagger.hilt.components.SingletonComponent
+import javax.inject.Qualifier
+import javax.inject.Singleton
+
+/**
+ * Hilt module that provides [LocationService]
+ */
+@Module
+@InstallIn(SingletonComponent::class)
+abstract class LocationModule {
+
+    @Qualifier
+    @Retention(AnnotationRetention.RUNTIME)
+    internal annotation class LocationProvider
```

Comment:
What is this annotation used for ? I never saw this in Hilt

Reply:
I use it to tag a provide method (L.46-51) that it provides a LocationProvider. It is necessary because LocationProvider is a String and to inject a LocationProvider, the String type is not enough for hilt. I hope it's clear.